actionpack/lib/action_controller/metal/renderers.rb

```diff
@@ -16,13 +16,65 @@ def initialize(format)
       super "No renderer defined for format: #{format}"
     end
   end
+  # RFC: Consider documenting or testing MissingRenderer
+  #
+  # See <tt>Responder#api_behavior</tt>
+  # https://github.com/plataformatec/responders/blob/f0e093d01e799d8b67952c013daf8808e9ea02db/lib/action_controller/responder.rb#L209
+  #
+  #   # This is the common behavior for formats associated with APIs, such as :xml and :json.
+  #   def api_behavior
+  #     raise MissingRenderer.new(format) unless has_renderer?
+  #   end
+  #
+  #   # Check whether the necessary Renderer is available
+  #   def has_renderer?
+  #     Renderers::RENDERERS.include?(format)
+  #   end
```

Comment:
I don't think there's really an easy way to do implement `MissingRenderer` at this time. Given this file defines


```ruby
     def render_to_body(options)
        _render_to_body_with_renderer(options) || super
      end
```

When there's no registered renderer,`_render_to_body_with_renderer(options)` is `nil` and the method calls `super`, which may raise `ActionView::MissingTemplate` in an ActionController::Base, but just be nil in an ActionController::Metal. So, if we consider a renderer an `API` concern, in the language of the [`Responders` gem](https://github.com/plataformatec/responders/blob/f0e093d01e799d8b67952c013daf8808e9ea02db/lib/action_controller/responder.rb#L209), how do we know which `content_type` or `Mime::Type.lookup(content_type)` should be a missing template (html/plain/text) vs. missing renderer (json/xml/js/csv)?